broker-daemon/utils/migrate-store.spec.js

```diff
@@ -1,117 +1,192 @@
-  })
+      reverts = []
+      reverts.push(migrateStore.__set__('promisify', promisifyStub))
+    })
 
-  it('flushes the batches on end', async () => {
-    await migrateStore(sourceStore, targetStore, createDbOperation, 1)
+    afterEach(() => {
+      reverts.forEach(r => r())
+    })
 
-    expect(targetStore.batch).to.have.been.calledTwice()
-    expect(targetStore.batch).to.have.been.calledWith(sinon.match.array.deepEquals(fakeOperations.slice(0, 2)))
-    expect(targetStore.batch).to.have.been.calledWith(sinon.match.array.deepEquals(fakeOperations.slice(2)))
-  })
+    it('runs the KV through a db operation', async () => {
+      migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+      const onData = onStub.args[0][1]
+      await onData(params)
+      expect(createDbOperation).to.have.been.calledWith(params.key, params.value)
+    })
 
-  it('skips empty db operations', async () => {
-    fakeOperations.push(undefined)
+    it('returns if the db operation is null', async () => {
+      createDbOperation.returns(null)
+      migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+      const onData = onStub.args[0][1]
+      await onData(params)
+      expect(batchStub).to.not.have.been.called()
+    })
 
-    await migrateStore(sourceStore, targetStore, createDbOperation)
+    it('pushes the db operation onto the batch', async () => {
+      migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+      const onData = onStub.args[0][1]
+      createDbOperation.returns(null)
+      await onData(params)
+      createDbOperation.returns(records[0])
+      await onData(params)
+      expect(batchStub.args[0][0].length).to.be.eql(1)
+    })
 
-    expect(targetStore.batch).to.have.been.calledWith(sinon.match.array.deepEquals(fakeOperations.slice(0, fakeOperations.length - 1)))
+    it('returns if the batch size has not been hit', async () => {
+      migrateStore(sourceStore, targetStore, createDbOperation, 2)
+      const onData = onStub.args[0][1]
+      await onData(params)
+      expect(batchStub).to.not.have.been.called()
+    })
+
+    it('pauses the stream before processing a batch', async () => {
+      migrateStore(sourceStore, targetStore, createDbOperation, 1)
+      const onData = onStub.args[0][1]
+      await onData(params)
+      expect(pauseStub).to.have.been.calledBefore(batchStub)
+    })
+
+    it('resumes the stream after processing the batch', async () => {
+      migrateStore(sourceStore, targetStore, createDbOperation, 1)
+      const onData = onStub.args[0][1]
+      await onData(params)
+      expect(batchStub).to.have.been.calledBefore(resumeStub)
+    })
+
+    context('batch is full', () => {
+      it('awaits the previous batch if the batch size is large', async () => {
+        migrateStore(sourceStore, targetStore, createDbOperation, 5)
+        const onData = onStub.args[0][1]
+        await onData(params)
+        expect(batchStub).to.not.have.been.called()
```

Comment:
This seems quite repetitive. Can't you just call `onData(params)` 5 times in a row using a loop and then check that `batchStub` has not been called? Not sure you gain much checking after each go.

Also - seems like 2 or 3 would get the point across just as well as 5.